packages/wallet/ui/src/util/WalletBackendAdapter.js

```diff
@@ -139,36 +142,74 @@ export const makeWalletBridgeFromFollower = (
   );
 
   // We assume just one cosmos purse per brand.
+  /** @type {Record<number, import('@agoric/smart-wallet/src/offers.js').OfferStatus & {status: 'accept' | 'rejected'}>} */
   const offers = {};
+  /**
+   * @typedef {{
+   *  brand?: Brand,
+   *  brandPetname?: Petname,
+   *  currentAmount: Amount,
+   *  pursePetname?: Petname,
+   *  displayInfo?: DisplayInfo,
+   * }} PurseInfo
+   * @type {Map<Brand, PurseInfo>}
+   */
   const brandToPurse = new Map();
+  /** @type {Map<Petname, Brand>} */
   const pursePetnameToBrand = new Map();
 
+  if (firstCallback) {
+    firstCallback();
+    Object.values(notifierKits).forEach(({ updater }) =>
+      updater.updateState([]),
+    );
+    firstCallback = undefined;
+  }
+
   const updatePurses = () => {
     const purses = [];
     for (const [brand, purse] of brandToPurse.entries()) {
       if (purse.currentAmount && purse.brandPetname) {
+        assert(purse.pursePetname, 'missing purse.pursePetname');
         pursePetnameToBrand.set(purse.pursePetname, brand);
         purses.push(purse);
       }
     }
     notifierKits.purses.updater.updateState(harden(purses));
   };
 
-  const followLatest = async () => {
-    const firstHeight = await getFirstHeight(follower);
+  const fetchCurrent = async () => {
+    console.log('fetchCurrent()');
+    const latestIterable = await E(currentFollower).getLatestIterable();
+    const iterator = latestIterable[Symbol.asyncIterator]();
+    const latest = await iterator.next();
```

Comment:
To prevent a race between the two followers, we need to store the height at which `latest` was published, like the way we used to get from `getFirstHeight`.